server/router.js

```diff
@@ -30,6 +30,7 @@ api.use(['/defenses'], isLoggedIn)
 api.get('/users/', users.List)
 api.get('/users/:id', users.Show)
 api.post('/users/', bodyJson, users.Create)
+api.put('/users/:id', bodyJson, users.Update)
```

Comment:
Testa que o professor pode fazer isso.

Reply:
Spoiler: não

Reply:
Ler linha 26

Reply:
Criar também?

Reply:
Você vai ter que tirar o '/users' da linha 26 pois isso força ser admin.

Em seguida, nas respectivas rotas do usuário você coloca `isLoggedIn` e `isAdmin` one precisar e apenas `isLoggedIn` no caso do update.

Reply:
Ah sim, entendi.
Pra criar, somente o `admin`, já pra atualizar, o `teacher` pode. Estava bugado já

Reply:
Acho q devo refazer o Update.
Tipo, o professor só poderá atualizar sua própria senha.
Assim, da forma que está, um professor pode atualizar a senha de qualquer professor.
Confere?

Reply:
Sim

Reply:
oh god

Reply:
Pode ver se é isso?

Reply:
Não há mudanças no router.

Reply:
Aonde estão as mudanças no users.Update?

Reply:
Isso é estranho.
Pq eu modifiquei uhsushush